Foundation/NSData.swift

```diff
@@ -414,6 +421,39 @@ open class NSData : NSObject, NSCopying, NSMutableCopying, NSSecureCoding {
             free(buffer)
         }
     }
+
+#if os(Linux)
+    internal static func readZeroSizeFile(_ fd: Int32) throws -> NSDataReadResult {
+        let blockSize = 1024 * 1024 // 1MB
+        var data: UnsafeMutableRawPointer? = nil
+        var bytesRead = 0
+        var amt = 0
+
+        repeat {
+            data = realloc(data, bytesRead + blockSize)
```

Comment:
This may return NULL, in which case the unwrap subsequently will fail. Is there any reason why we can't use a single byte array for reading and then have an NSMutableData to append the bytes into? That would solve the realloc calls.